apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/TxnHistoryTable.tsx

```diff
@@ -1,153 +1,199 @@
-import { useEffect, useState } from 'react';
+import { SubGraphResponse } from 'hooks/core/types';
+import { useCallback, useEffect, useState } from 'react';
 import styled from 'styled-components';
+import { fetchGenericSubgraph } from 'utils/subgraph';
+import { TxHistoryFilterType } from '.';
+import { DashboardType } from '../DashboardContent';
+import { format } from 'date-fns';
+import { TableRow, TxnDataTable } from './TxnDataTable';
+import { PaginationControls } from './TxnPaginationControl';
+import env from 'constants/env/local';
 
-type TableRow = {
-  date: string;
-  debit: number;
-  credit: number;
+type Props = {
+  dashboardType: DashboardType;
+  filter: TxHistoryFilterType;
 };
 
-type Props = {
-  filter: string;
+type StrategyType = 'TlcStrategy' | 'DsrBaseStrategy' | 'RamosStrategy' | 'TempleBaseStrategy' | 'All';
+
+type Transactions = {
+  hash: string;
+  amount: string;
+  amountUsd: string;
+  id: string;
+  from: string;
+  to: string;
+  kind: 'Repay' | 'Borrow';
+  timestamp: string;
+}[];
+
+type StrategyTxns = {
+  name: StrategyType;
+  id: string;
+  transactionCount: number;
+  transactions: Transactions;
+}[];
+
+type Meta = {
+  block: {
+    number: number;
+  };
 };
 
-const TxnHistoryTable = ({ filter }: Props) => {
+type FetchTxnsResponse = SubGraphResponse<{ strategies: StrategyTxns; _meta: Meta }>;
+
+const TxnHistoryTable = ({ dashboardType, filter }: Props) => {
   const [currentPage, setCurrentPage] = useState(1);
   const [rowsPerPage, setRowsPerPage] = useState(10);
   const [data, setData] = useState<TableRow[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [blockNumber, setBlockNumber] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+
+  const restartTable = useCallback(async()=> {
+    setData([]);
+    setCurrentPage(1);
+    setBlockNumber(0);
+    setTotalPages(0);
+  }, []);
+  // Set default values TotalPages & BlockNumber on start
+  useEffect(() => {
+    (async ()=> {
+      await restartTable();
+      const strategyType = dashboardTypeToStrategyType(dashboardType);
+      const strategyQuery = strategyType === 'All' ? `` : `( where: {name: "${strategyType}"} )`;
+      // TODO: workout how to update transactionCount give the filter
+
+      const { data: res } = await fetchGenericSubgraph<FetchTxnsResponse>(
+        env.subgraph.templeV2,
+        `{
+            strategies${strategyQuery} {
+              transactionCount
+            }
+            _meta {
+              block {
+                number
+              }
+            }
+          }`
+      );
```

Comment:
@medariox would be possible to have a dynamic `transactionCount` from subgraph given the `transactions` each strategy has with the filter `All`, `Last 30 days` or `Last week`, e.g. [this query](https://api.studio.thegraph.com/query/520/v2-sepolia/version/latest/graphql?query=%7B%0A++strategies%28where%3A+%7B+name%3A+%22TlcStrategy%22+%7D%29+%7B%0A++++name%0A++++id%0A++++transactionCount%0A++++transactions%28orderBy%3A+timestamp%2C+orderDirection%3A+desc+where%3A+%7Btimestamp_gt%3A+1695101129%7D%29+%7B%0A++++++id%0A++++++hash%0A++++++timestamp%0A++++++kind%0A++++%7D%0A++%7D%0A++_meta+%7B%0A++++block+%7B%0A++++++number%2C%0A++++%7D%0A++%7D%0A%7D) ideally would return `transactionCount = 3`

Reply:
That would add too much overhead to the subgraph code. In my opinion, the way to go here is to just count the returned `Transaction` entities in the frontend code.

Reply:
ok thx, I've done calc in the frontend, not needing anymore `transactionCount` in subgraph

Reply:
> ok thx, I've done calc in the frontend, not needing anymore `transactionCount` in subgraph

Are you sure you don't need it? How are you going to do the pagination after the returned entities limit has been reached (default 100, max 1000)? Suppose you want to show the total amount of pages for all transactions, and you limit your query returns to only the first 10 (with `first: 10`). You can only count the 10, but you still need `transactionCount` to paginate properly. See [this example query](https://api.studio.thegraph.com/query/520/v2-sepolia/version/latest/graphql?query=%7B%0A++strategies%28where%3A+%7Bname%3A+%22TlcStrategy%22%7D%29+%7B%0A++++name%0A++++id%0A++++transactionCount%0A++++debtTokens+%7B%0A++++++name%0A++++++id%0A++++++symbol%0A++++%7D%0A++++transactions%28first%3A+10%2C+orderBy%3A+timestamp%2C+orderDirection%3A+desc%29+%7B%0A++++++amount%0A++++++amountUSD%0A++++++from%0A++++++to%0A++++++hash%0A++++++kind%0A++++%7D%0A++%7D%0A++_meta+%7B%0A++++block+%7B%0A++++++number%0A++++%7D%0A++%7D%0A%7D).

Reply:
I'm currently doing two queries, one to get the initial total transaction count (sum all the strategy transaction lengths), which is the one used to get the total number of pages, and the other query has all the required filters applied.

The above works fine but I didn't know about the default max 1000 so will prob need some extra help. The problem with the `transactionCount` you've implemented is that I need to apply on top of that the filter `All`, `Last 30 days` or `Last week`, otherwise it'll show always the total number of pages, which would cause glitches on the ui, e.g. `TlcStrategy` on the last week only has 2 transactions, but the `transactionCount` is equal to 17, [see query](https://api.studio.thegraph.com/query/520/v2-sepolia/version/latest/graphql?query=%7B%0A++strategies+%7B%0A++++name%0A++++id%0A++++transactionCount%0A++++transactions%28orderBy%3A+timestamp%2C+orderDirection%3A+desc%2C+where%3A+%7Btimestamp_gt%3A1697572322%7D%29+%7B%0A++++++token+%7B%0A++++++++name%0A++++++++id%0A++++++++symbol%0A++++++%7D%0A++++++amount%0A++++++amountUSD%0A++++++from%0A++++++to%0A++++++hash%0A++++++kind%0A++++%7D%0A++%7D%0A++_meta+%7B%0A++++block+%7B%0A++++++number%0A++++%7D%0A++%7D%0A%7D)

maybe we can have a chat a sorted it out together?